Rework home page (#4855)
OttoWinter/esphomedocs: index.rst

            }));
            
            // Add results component
            instance.add(new PagefindModularUI.ResultList({
                containerElement: "#nav-search-results"
            }));
            
            // Show/hide results
            instance.on("results", (results) => {
                if (results.results.length) {
                    resultsContainer.style.display = 'block';
                } else {
                    resultsContainer.style.display = 'none';
                }
            });
            
            // Hide results when clicking outside
            document.addEventListener('click', (event) => {
                if (!event.target.closest('.nav-search')) {
                    resultsContainer.style.display = 'none';
                }
            });
            
            // Focus input on click
            searchInput.addEventListener('click', (event) => {
                event.stopPropagation();
            });
        });
    </script>

.. raw:: html
    :file: images/logo-text.svg

.. raw:: html

    <div class="hero-container">
        <div class="hero-content">
            <h1>Smart Home Made Simple</h1>

>             <p>Turn your ESP8266/ESP32 or RP2040 boards into powerful smart home devices with simple YAML configuration</p>

Reviewer: ```suggestion
            <p>Turn your ESP32 , ESP8266 or RP2040 boards into powerful smart home devices with simple YAML configuration</p>
```

        </div>
        <div class="hero-image">
            <img src="_images/hero.png" alt="ESPHome devices" />
        </div>
    </div>

.. _what-is-esphome:

What is ESPHome?
================

ESPHome is an open-source firmware framework that simplifies the process of creating custom firmware for popular WiFi-enabled microcontrollers. With ESPHome, you can:

* **Create custom smart home devices** using simple YAML configuration files
* **Integrate seamlessly with Home Assistant** for a unified smart home experience
* **Control and monitor** your devices through multiple interfaces (web, API, MQTT)
* **Automate your home** with powerful on-device automations

> * **Update your devices wirelessly** (OTA) without physical access

Reviewer: ```suggestion
* **Update your devices wirelessly** "over the air" (OTA) without physical access
```

        <div class="feature-card">
            <div class="feature-icon"><i class="fas fa-wifi"></i></div>
            <h3>Wireless Updates</h3>
            <p>Update your devices over-the-air without physical access</p>
        </div>
        <div class="feature-card">
            <div class="feature-icon"><i class="fas fa-puzzle-piece"></i></div>
            <h3>Modular Design</h3>
            <p>Support for hundreds of sensors, displays, and other components</p>
        </div>
        <div class="feature-card">
            <div class="feature-icon"><i class="fas fa-shield-alt"></i></div>
            <h3>Local Control</h3>
            <p>Devices work locally without cloud dependencies</p>
        </div>
    </div>

.. _who-uses-esphome:

Who Uses ESPHome?
=================

ESPHome is used by a diverse community of smart home enthusiasts, makers, and professionals:

**DIY Enthusiasts**
    Create custom sensors, switches, and displays tailored to specific needs

**Smart Home Hobbyists**
    Extend their home automation systems with affordable custom devices

**Professional Integrators**
    Deploy reliable, locally-controlled smart devices for clients

**Manufacturers**

>     Create "Made for ESPHome" certified products with standardized firmware

Reviewer: Maybe we inline a link to the M4E doc?

.. _getting-started:

Getting Started
===============

Getting started with ESPHome is easy. Choose the method that works best for you:

.. raw:: html

    <div class="getting-started-grid">
        <div class="getting-started-card">
            <h3>From Home Assistant</h3>
            <p>The easiest way to get started with ESPHome is through the Home Assistant add-on.</p>
            <ol>
                <li>Open Home Assistant</li>
                <li>Go to Settings → Add-ons → Add-on Store</li>
                <li>Find and install the ESPHome add-on</li>

>                 <li>Open the ESPHome dashboard and create your first device</li>

Reviewer: ```suggestion
                <li>Open the ESPHome Device Builder and create your first device</li>
```

            <ol>
                <li>Install Python 3.11 or higher</li>
                <li>Install ESPHome: <code>pip install esphome</code></li>
                <li>Create a configuration file</li>
                <li>Compile and upload: <code>esphome run my_config.yaml</code></li>
            </ol>
            <a href="/guides/getting_started_command_line.html" class="btn btn-primary">Detailed Instructions</a>
        </div>
        <div class="getting-started-card">
            <h3>Ready-Made Projects</h3>
            <p>Start with a pre-configured project for common use cases.</p>
            <ol>
                <li>Browse the project library</li>
                <li>Select a project that matches your hardware</li>
                <li>Follow the installation instructions</li>
                <li>Customize as needed</li>
            </ol>
            <a href="/projects/" class="btn btn-primary">Browse Projects</a>
        </div>
    </div>

Basic Configuration Example
---------------------------

Here's a simple example of an ESPHome configuration file:

.. code-block:: yaml

    # Basic configuration for an ESP32 device
    esphome:

>       name: living_room_sensor

Reviewer 2: ```suggestion
      name: living-room-sensor
```

      friendly_name: Living Room Sensor

    # Hardware configuration
    esp32:
      board: esp32dev
      framework:

>         type: arduino

Reviewer 2: ```suggestion
        type: esp-idf
```

    # Enable Home Assistant API

>     api:
>       encryption:
>         key: "YOUR_ENCRYPTION_KEY"
> 
>     # Enable over-the-air updates
>     ota:
>       password: "YOUR_OTA_PASSWORD"
> 
>     # Connect to WiFi
>     wifi:
>       ssid: !secret wifi_ssid
>       password: !secret wifi_password
>       
>       # Enable fallback hotspot if WiFi connection fails
>       ap:
>         ssid: "Living Room Sensor Fallback"
>         password: "YOUR_FALLBACK_PASSWORD"

Reviewer 2: Lets simplify a bit
```suggestion
    api:

    # Enable over-the-air updates
    ota:
      - platform: esphome

    # Connect to WiFi
    wifi:
      ssid: !secret wifi_ssid
      password: !secret wifi_password
      
      # Enable fallback hotspot if WiFi connection fails
      ap:
```

    # Enable web server
    web_server:

>       port: 80

Reviewer: Continuing with simplification -- `port` is optional, I think we could eliminate it here.
```suggestion
```

    # Add a temperature sensor
    sensor:
      - platform: dht
        pin: GPIO15
        temperature:

>           name: "Living Room Temperature"
>         humidity:
>           name: "Living Room Humidity"

Reviewer 2: Friendly name will be prepended by Home Assistant
```suggestion
          name: "Temperature"
        humidity:
          name: "Humidity"
```

.. _using-with-home-assistant:

Using with Home Assistant
=========================

ESPHome and Home Assistant are designed to work together seamlessly, providing a powerful smart home ecosystem.

Automatic Discovery
-------------------

Home Assistant automatically discovers ESPHome devices on your network. When you add a new ESPHome device:

1. Make sure your device is connected to the same network as Home Assistant
2. Home Assistant will show a notification when it discovers the new device
3. Click "Configure" to add it to your system
4. The device and all its entities will appear in your Home Assistant dashboard

Dashboard Integration
---------------------

ESPHome entities appear in Home Assistant just like any other device:

* **Sensors** show up in the appropriate cards and graphs
* **Switches and lights** can be controlled directly
* **Binary sensors** can trigger automations
* **Climate devices** integrate with the thermostat controls

Automations
-----------

You can create powerful automations using ESPHome devices in Home Assistant:

* Use sensor readings to trigger actions
* Control ESPHome devices based on conditions
* Include ESPHome devices in scenes and scripts

> * Create complex multi-device automations

Reviewer: I worry this leads people to think they can get ESPHome devices talking to each other directly...and we're not there just yet. haha
```suggestion
* Create complex automations based on a number of inputs and/or conditions
```

Author: It does say "automations using ESPHome devices _in Home Assistant_"

Advanced Features
-----------------

> * **ESPHome Dashboard Add-on**: Manage all your ESPHome devices directly from Home Assistant

Reviewer: ```suggestion
* **ESPHome Device Builder Add-on**: Manage all your ESPHome devices directly from Home Assistant
```

> * **YAML Synchronization**: Keep your configurations in sync with Home Assistant

Reviewer: I don't understand this point. haha

Author: It probably only makes sense to an AI.
